generators/upgrade/index.js

```diff
@@ -109,24 +132,42 @@ module.exports = UpgradeGenerator.extend({
         detectCurrentBranch: function() {
             var done = this.async();
             this.gitExec(['rev-parse', '-q', '--abbrev-ref', 'HEAD'], function(code, msg, err) {
-                if (code !== 0) this.error('Unable to detect current git branch:\n' + err);
+                if (code !== 0) this.error('Unable to detect current git branch:\n' + msg + ' ' + err);
                 this.sourceBranch = msg.replace('\n','');
                 done();
             }.bind(this));
         },
 
         prepareUpgradeBranch: function() {
             var done = this.async();
-            var createUpgradeBranch = function(callback) {
-                this.gitExec(['branch', '-q', UPGRADE_BRANCH], function(code, msg, err) {
-                    if (code !== 0) this.error('Unable to create ' + UPGRADE_BRANCH + ':\n' + err);
+            var recordCodeHasBeenGenerated = function() {
+                this.gitExec(['merge', '--strategy=ours', '-q', '--no-edit', UPGRADE_BRANCH], function(code, msg, err) {
+                    if (code !== 0) this.error('Unable to record current code has been generated with version ' +
+                        this.currentVersion + ':\n' + msg + ' ' + err);
+                    this.log('Current code recorded as generated with version ' + this.currentVersion);
+                    this._gitCheckout(UPGRADE_BRANCH, done);
+                }.bind(this));
+            }.bind(this);
+
+            var regenerate = function() {
+                this._regenerate(this.currentVersion, function() {
```

Comment:
not exactly, it's a real commit, but block-merged on master in order to tell Git master's HEAD is the HEAD of everything, including jhipster_upgrade stuff